src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write code here
+const list = document.body.children[1].children;
+const array = [...list].map(item => [
+  item.innerText,
+  item.dataset.position,
+  item.dataset.salary.replace(/[^0-9]/g, ''),
+  item.dataset.age,
+]);
+
+function sortList(arr) {
+  return arr.sort((a, b) => b[2] - a[2]);
+}
+
+function getEmployees(arr) {
+  return arr.map(item => {
+    return {
+      name: item[0],
+      position: item[1],
+      salary: '$' + item[2],
+      age: item[3],
+    };
+  });
+}
+
+sortList(array);
+
+const employees = getEmployees(array);
+const replace = document.querySelector('ul');
```

Comment:
```suggestion
const list = document.querySelector('ul');
```
replace не отображает что у тебя в этой переменной хранится ссылкан а дом элемент списка